### DISCUSSION TITLE
ZMQ Error due to port issues with lab mate

### DISCUSSION THREAD

**Discussion by henryplamondon** [talmolab/sleap#1751] [Help!] [answered]
I'm currently using SLEAP on a lab computer. Occasionally when I attempt to train a model, I receive a 

## "ZMQError: Address already in use"

If the process is another SLEAP process, I can just kill it using sudo kill -9. However, occasionally it's a lab mate of mine running a simple notebook using VS Code elsewhere in the lab. 
### I'm wondering if it's more efficient to change the ports from 9001 and 9002 in the SLEAP code and hardcore them to other ports to reroute from the SLEAP side or if it would be a more efficient use of my time to find a way to reroute my lab mate's usage of VS Code when running his ipynb's. I'm also wondering if changing the ports from the SLEAP code will introduce further downstream issues. 
Thanks in advance for the help!

**Comment by Marielamend** [comment]
Hello @henryplamondon,

Sorry for the delayed response!

You can try going to your SLEAP installation model folder > training_profiles > choose the specific model/s you want to use > set:
`"subscribe_to_controller":
`
`"publish_updates":`
_to false_:
`"subscribe_to_controller": false,
`
`"publish_updates": false,
`

Then you can test this by running training through the command line using the [sleap-train](https://sleap.ai/guides/cli.html?highlight=sleap%20train#sleap-train:~:text=Training-,sleap%2Dtrain,-%23) command and see how that goes!


Keep us updated!

Best,
Mariela

**Reply by henryplamondon** [reply]
Hi @Marielamend,

Thank you for the response! 

I've tried this and it works! However, the workflow of labeling and training is now quite different as I cannot use the GUI since I have to export the training package and manually change these options each time. This blocks me from visualizing predictions during training and seeing the loss curves as well as prevents me from using prediction-assisted labeling. Is there a way to systematically change the way the GUI implements these options so I don't have to continually export the training packages and change the file each time when I begin training a new model? Alternatively, is there a way to simplify the workflow of using the command line for training and visualization of models? 

Thank you again for your help in this process! SLEAP has been an extremely useful tool for me!

Best,
Henry

**Reply by eberrigan** [reply, marked as answer]
Hi @henryplamondon, 

This feature has been added in our [pre-release](https://github.com/talmolab/sleap/releases/tag/v1.4.1a1). Please make a new sleap environment with the updated conda package to use the new feature.

Thanks,

Elizabeth
